pkg/controllers/user/alert/configsyncer/configsyncer.go

```diff
@@ -512,3 +528,29 @@ func includeProjectMetrics(projectAlerts []*v3.ProjectAlertRule) bool {
 	}
 	return false
 }
+
+func getClusterAlertGroupBy(spec v3.ClusterAlertRuleSpec) []model.LabelName {
+	if spec.EventRule != nil {
+		return eventGroupBy
+	} else if spec.SystemServiceRule != nil {
+		return systemServiceGroupBy
+	} else if spec.NodeRule != nil {
+		return nodeGroupBy
+	} else if spec.MetricRule != nil {
+		return metricGroupBy
+	}
+
+	return nil
+}
+
+func getProjectAlertGroupBy(spec v3.ProjectAlertRuleSpec) []model.LabelName {
+	if spec.PodRule != nil {
+		return podGroupBy
```

Comment:
If you don't need to reuse the variable `podGroupBy `, just return a new array. Same as all returns in this function.